lib/edge_impulse/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Prepare Edge Impulse library zip archive
+if(CONFIG_EDGE_IMPULSE_BUILD_FROM_ZIP)
+
+  if(CONFIG_EDGE_IMPULSE_ZIP_PATH STREQUAL "")
+    message(FATAL_ERROR "Edge Impulse archive path must be specified in Kconfig.")
+  endif()
+
+  set(EI_ZIP_PATH ${APPLICATION_SOURCE_DIR}/${CONFIG_EDGE_IMPULSE_ZIP_PATH})
+  if (NOT EXISTS ${EI_ZIP_PATH})
+    message(FATAL_ERROR "Specified Edge Impulse archive (${EI_ZIP_PATH}) does "
+             "not exist. Download the archive manually and place it under "
+             "defined path or use CONFIG_EDGE_IMPULSE_BUILD_FROM_URI Kconfig "
+             "option to let the build system download the library "
+             "automatically.")
+  else()
+    message(STATUS "Using Edge Impulse archive: ${EI_ZIP_PATH}.")
+  endif()
+
+elseif(CONFIG_EDGE_IMPULSE_BUILD_FROM_URI)
+
+  if(CONFIG_EDGE_IMPULSE_DOWNLOAD_URI STREQUAL "")
+    message(FATAL_ERROR "Edge Impulse download URI must be specified in Kconfig.")
+  endif()
+
+  set(EI_ZIP_PATH ${CMAKE_BINARY_DIR}/ei_src.zip)
+
+  if(NOT EXISTS ${EI_ZIP_PATH})
+    # Start download, use API key that was provided.
+    message(STATUS "Downloading Edge Impulse files, please wait...")
+
+    if(DEFINED EI_API_KEY_HEADER)
+      file(DOWNLOAD ${CONFIG_EDGE_IMPULSE_DOWNLOAD_URI} ${EI_ZIP_PATH}
+           SHOW_PROGRESS
+           LOG download_log
+           STATUS download_status
+           HTTPHEADER "Accept: application/zip"
+           HTTPHEADER ${EI_API_KEY_HEADER})
+    else()
+      file(DOWNLOAD ${CONFIG_EDGE_IMPULSE_DOWNLOAD_URI} ${EI_ZIP_PATH}
+           SHOW_PROGRESS
+           LOG download_log
+           STATUS download_status
+           HTTPHEADER "Accept: application/zip")
+    endif()
+
+    list(GET download_status 0 status_code)
+    if (status_code EQUAL 0)
+      message(STATUS "Successfully downloaded Edge Impulse archive.")
+    else()
+      file(REMOVE ${EI_ZIP_PATH})
+      message(FATAL_ERROR "Failed to download Edge Impulse files. Please make "
+               "sure that CONFIG_EDGE_IMPULSE_DOWNLOAD_URI is properly defined "
+               "and EI_API_KEY_HEADER is specified (if needed).")
+    endif()
+  else()
+    message(STATUS "Edge Impulse zip archive is already downloaded.")
+  endif()
+
+else()
+  message(FATAL_ERROR "Unknown Edge Impulse build source.")
+endif()
+
+# Unzip Edge Impulse archive
+set(EI_BASE_DIR ${CMAKE_BINARY_DIR}/ei_src_unpacked)
+
+if (NOT EXISTS ${EI_BASE_DIR})
+  file(MAKE_DIRECTORY ${EI_BASE_DIR})
+endif()
+
+execute_process(
```

Comment:
Please make sure that cmake is rerun in case this package changes. And make sure package is updated when config options are changed.

Reply:
done

Reply:
I also added CMake required version and used new API to extract the zip archive